actors/builtin/miner/miner_actor.go

```diff
@@ -383,14 +326,20 @@ func (a Actor) ProveCommitSector(rt Runtime, params *ProveCommitSectorParams) *a
 			rt.Abort(exitcode.ErrIllegalState, "failed to delete precommit for sector %v: %v", sectorNo, err)
 		}
 
-		st.ProvingSet = st.ComputeProvingSet()
+		// Do not update proving set during challenge window
+		if !st.inChallengeWindow(rt) {
+			st.ProvingSet = st.Sectors
+		}
```

Comment:
Rather than do this every time, how about copying sectors to proving set only/exactly when the challenge window opens?

I don't think the proving set is relevant for EPost - that just uses the committed set (except faults).

Reply:
we don't have anything that actually happens when the window opens, and this assignment is free (we have to rewrite the entire struct anyways)